pkg/controller/machine/controller.go

```diff
@@ -434,7 +435,16 @@ func (r *ReconcileMachine) setPhase(machine *machinev1.Machine, phase string, er
 		}
 
 		// Since we may have mutated the local copy of the machine above, we need to calculate baseToPatch here.
+		// Any updates to the status must be done after this point.
 		baseToPatch := client.MergeFrom(machine.DeepCopy())
+
+		if phase == phaseFailed {
```

Comment:
shouldn't this reuse the if block in line 429?

Reply:
If we use the block from above, then it's before the `baseToPatch` is created and as such, the patching mechanism thinks that the changes were there already. This modification has to come after `baseToPatch` is created, but the other one has to come before it, because it is modifying the spec.

I added the line on 438 to try and make this more obvious